ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php

```diff
@@ -315,10 +315,20 @@ private function rewrite(array $column, array $data): string {
         while (count($keys) > 1) {
           $level = array_shift($keys);
           $parent[$level] = $parent[$level] ?? [];
+          if (is_scalar($parent[$level])) {
+            $parent[$level] = [
+              'id' => $parent[$level],
```

Comment:
Last week you said that not all fields ending in `_id` are FKs. I'm guessing that:
* All keys that *are* FKs end in `_id`;
* We avoid the non-FK IDs because of `is_scalar`.

If both of those are true, this looks good.

Reply:
@MegaphoneJon there's a difference between field name and target field. At least in the context of FKs they mean:
- **Field name** name of the field we're linking *from*.
- **Target field** name of the field we're linking *to*.

Example: field name `employer_id` is an FK to `civicrm_contact` with target field `id`.

I think field name always ends in ID in the core schema but I try to avoid making that assumption because extensions can do what they want. But that's ok this PR doesn't care about the field name. It's just appending the name of the target field, which is reliably always "id".

If the field name were something else, that would be ok too, as this just appends `.id` to the end regardless of the name.